rust/ip-packet/src/tcp_header_slice_mut.rs

```diff
@@ -13,6 +13,14 @@ impl<'a> TcpHeaderSliceMut<'a> {
         Ok(Self { slice })
     }
 
+    pub fn get_source_port(&self) -> u16 {
+        u16::from_be_bytes([self.slice[0], self.slice[1]])
+    }
+
+    pub fn get_destination_port(&self) -> u16 {
+        u16::from_be_bytes([self.slice[2], self.slice[3]])
+    }
```

Comment:
I don't think we should be _reading_ data using a `-Mut` slice. To read the data, can we use the immutable slice from `etherparse`?

Reply:
If I read from the immutable slice, I'd need to first parse as immutable, store the fields, drop that reference then re-parse to the mutable slice and modify, that seems cumbersome. Normally mutable structs also allows you to read the fields.